### DISCUSSION TITLE
How to bundle deck.gl in a library?

### DISCUSSION THREAD

**Discussion by jeffpeck10x** [visgl/deck.gl#6879] [Q&A]
I tried to bundle deck.gl in a library using parcel and it builds, but at runtime, I get an error due to this line in probe.gl:
https://github.com/uber-web/probe.gl/blob/master/modules/log/src/utils/hi-res-timestamp.ts#L10
```
  } else if ('hrtime' in process) {
```

complaining that `process` is undefined. (It technically shouldn't even hit that line because the condition before it `isBrowser && 'performance' in window` should be true at runtime, so that is odd too.)

---

Next, I tried to bundle it with rollup. That won't even build and this is the error:
```
[!] Error: 'global' is not exported by ../../node_modules/probe.gl/env.js, imported by ../../node_modules/@deck.gl/core/dist/esm/lib/init.js
https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module
../../node_modules/@deck.gl/core/dist/esm/lib/init.js (3:9)
```

That issue was also reported here back in January, 2021 with no reponse:
https://github.com/visgl/deck.gl/discussions/5299

---

I also found this attempt with a much older version of deck.gl, from 2019, but this too does not build:
https://github.com/danmarshall/deckgl-rollup

---

Has anyone been able to successfully package deck.gl? Should I try with webpack? Something else?

Or is it just some config parameter that I am missing?

**Comment by Pessimistress** [comment]
We are aware that Parcel breaks when a module exports anything named `global` or `process`, see https://github.com/visgl/deck.gl/issues/6351

I do not have any experience with rollup. We publish all the pre-built bundles with Webpack, and I've had success with esbuild as well.
